pkg/kubelet/kubelet.go

```diff
@@ -2052,17 +1978,17 @@ func getPodReadyCondition(spec *api.PodSpec, info api.PodInfo) []api.PodConditio
 }
 
 // GetPodStatus returns information from Docker about the containers in a pod
-func (kl *Kubelet) GetPodStatus(podFullName string, uid types.UID) (api.PodStatus, error) {
+func (kl *Kubelet) GetPodStatus(podFullName string) (api.PodStatus, error) {
 	// Check to see if we have a cached version of the status.
-	cachedPodStatus, found := kl.getPodStatusFromCache(podFullName)
+	cachedPodStatus, found := kl.statusSyncer.GetPodStatus(podFullName)
 	if found {
 		glog.V(3).Infof("Returning cached status for %q", podFullName)
 		return cachedPodStatus, nil
 	}
-	return kl.generatePodStatus(podFullName, uid)
+	return kl.generatePodStatus(podFullName)
 }
 
-func (kl *Kubelet) generatePodStatus(podFullName string, uid types.UID) (api.PodStatus, error) {
+func (kl *Kubelet) generatePodStatus(podFullName string) (api.PodStatus, error) {
```

Comment:
Eventually we should subsume this into the new status object too. Not for this PR though :) there is some more refactoring needed before that I think.